portfolio/src/main/java/com/google/sps/data/Book.java

```diff
@@ -8,18 +9,51 @@
 @AutoValue
 public abstract class Book {
   public static Builder builder() {
-    return new AutoValue_Book.Builder();
+    return new AutoValue_Book.Builder()
+        .categories(new ArrayList<String>())
+        .authors(new ArrayList<String>())
+        .language("N/A")
+        .description("N/A")
+        .infoLink("N/A")
+        .pageCount("N/A")
+        .publishedDate("N/A")
+        .publisher("N/A")
+        .maturityRating("N/A")
+        .thumbnail("N/A")
+        .isbn("N/A");
   }
 
   public abstract Builder toBuilder();
   public abstract String title();
   public abstract Set<String> genre();
   public abstract ImmutableList<String> reviews();
+  public abstract ImmutableList<String> categories();
+  public abstract ImmutableList<String> authors();
+  public abstract String language();
+  public abstract String description();
+  public abstract String infoLink();
+  public abstract String pageCount();
+  public abstract String publishedDate();
+  public abstract String publisher();
+  public abstract String maturityRating();
+  public abstract String thumbnail();
+  public abstract String isbn();
 
   @AutoValue.Builder
   public static abstract class Builder {
     public abstract Builder title(String title);
     public abstract Builder genre(Set<String> genre);
+    public abstract Builder categories(ArrayList<String> categories);
+    public abstract Builder authors(ArrayList<String> author);
+    public abstract Builder language(String language);
+    public abstract Builder description(String description);
+    public abstract Builder infoLink(String infoLink);
+    public abstract Builder pageCount(String pageCount);
```

Comment:
Should the pageCount be a integer?

Reply:
We could cast the variable but I don't think that's necessary. This is just going to get displayed later on as a string and we won't be interacting with is as an integer so I think keeping it as a string is fine for our purposes.

Reply:
The problem with keeping page count as String is the data-validation. If the data contains `pageCount=cat`,  you wouldn't know that it's invalid value.

In theory you could define everything as a String, but doing so would make it difficult to use those variables. I agree that the display use case doesn't need to interpret the page count value, but what about any future use cases, for example, search all books of page length 1-100. This could be useful information for recommendation engine, for example.

Reply:
That's a good point. I'll change the pageCount to an integer and once we have the end-to-end implementation working, I'll make sure that all the fields are being validated.